### PR TITLE
plugins/rustaceanvim: add lsp assertion

### PR DESCRIPTION
**Pull request by khaneliman**
Resolves https://github.com/nix-community/nixvim/issues/1258

### CODE REVIEW

**Review by GaetanLepage — APPROVED** [review]
Thanks

**Comment by khaneliman** [comment]
@mergifyio queue

File: plugins/languages/rust/rustaceanvim/default.nix [hunk 1]
```diff
@@ -56,6 +56,16 @@ helpers.neovim-plugin.mkNeovimPlugin {
 
         globals.rustaceanvim = cfg.settings;
 
+        assertions = [
+          {
+            assertion = cfg.enable -> !config.plugins.lsp.servers.rust-analyzer.enable;
```
Comment (MattSturgeon) [inline plugins/languages/rust/rustaceanvim/default.nix:61]: I don't think this does what you intended.

First of all, `cfg.enable` is always true within `extraConfig`. This means we're effectively only checking the rhs of the expression.

If we moved it out of `extraConfig` and into an imported module, then the assertion would be:

_"Assert that `cfg.enable` is false **or** `rust-analyzer.enable` is false"_

Which does sound correct based on your messages text.

**Review by MattSturgeon — CHANGES_REQUESTED** [review]

File: plugins/languages/rust/rustaceanvim/default.nix (hunk 1 above)
Comment (khaneliman) [reply plugins/languages/rust/rustaceanvim/default.nix:61]: I mean, the intended behavior is to not allow both enabled at the same time. If this gets enabled, we are checking the other isn’t also enabled. The lhs being redundant doesn’t change that, it was just to make it obvious what was going on

Comment (MattSturgeon) [reply plugins/languages/rust/rustaceanvim/default.nix:61]: Yeah you're right.

> it was just to make it obvious what was going on

Gotcha. Ironically, I think that's what threw me off :grinning:

**Pull request merged**